master/internal/config/config.go

```diff
@@ -357,6 +358,28 @@ func (t *TLSConfig) ReadCertificate() (*tls.Certificate, error) {
 type InternalConfig struct {
 	AuditLoggingEnabled bool                   `json:"audit_logging_enabled"`
 	ExternalSessions    model.ExternalSessions `json:"external_sessions"`
+	// DISCUSS: do we want to use a unix socket? change the impl on master and lore server.
```

Comment:
unix sockets:  higher throughput; better security; limited portability; inter-host complexity; fewer tools.
tcp sockets:  os portability; network communication; more debugging tools; lower performance;

IMO as long as we don't start proxying to a public interface the TCP socket is a good flexible route to go for our use.

Reply:
I would opt for portability in general, but I do wonder how much impact the throughput difference would make in the case of the upload api, since we very well may be uploading ~1GB files at max. Once we have a tangible e2e setup with this proxy, we can see how much of a difference that makes.